### PR TITLE
bybit: add new api endpoints

### PR DESCRIPTION
**Pull request by sc0Vu**
Bybit release new api endpoints in different section. In this PR, I'll add these into ccxt.

Change log:
* add withdraw and deposit endpoints
* add cross margin trading endpoints

### CODE REVIEW

File: js/bybit.js [hunk 1]
```diff
@@ -3564,6 +3584,99 @@ module.exports = class bybit extends Exchange {
...
+        //
+        //     {
+        //         chain_type: 'Arbitrum One',
+        //         address_deposit: '0x83a127952d266A6eA306c40Ac62A4a70668FE3BE',
+        //         tag_deposit: '',
+        //         chain: 'ARBI'
+        //     }
+        //
+        const address = this.safeString (depositAddress, 'address_deposit');
+        const tag = this.safeString (depositAddress, 'tag_deposit');
```
Comment (kroitor) [inline js/bybit.js:3597]: If the tag is an empty string, we should set it to `undefined`:

```suggestion
        let tag = this.safeString (depositAddress, 'tag_deposit');
        if (tag === '') {
            tag = undefined;
        }
```

File: js/bybit.js [hunk 2]
```diff
@@ -3564,6 +3584,102 @@ module.exports = class bybit extends Exchange {
...
+        //         chain: 'ARBI'
+        //     }
+        //
+        const address = this.safeString (depositAddress, 'address_deposit');
+        let tag = this.safeString (depositAddress, 'tag_deposit');
+        if (tag === '') {
+            tag = undefined;
+        }
+        const code = this.safeString (currency, 'code');
+        const chain = this.safeString (depositAddress, 'chain');
```
Comment (kroitor) [inline js/bybit.js:3602]: We must add the logic to parse the network here to unify the networks to the following:

- TRX > TRC20
- BSC > BSC
- ETH > TRC20

In order to solve this we will need to add this endpoint:

- https://bybit-exchange.github.io/docs/account_asset/#t-coin_info_query

And will have to implement the `networks` property in each currency:

- https://github.com/ccxt/ccxt/blob/master/js/huobi.js#L2420
- https://github.com/ccxt/ccxt/blob/master/js/huobi.js#L4064

Comment (sc0Vu) [reply js/bybit.js:3602]: The endpoint was added too. In this case one api is private and the other one is public, should we only call this when fetchDepositAddress?

Comment (kroitor) [reply js/bybit.js:3602]: > The endpoint was added too. In this case one api is private and the other one is public, should we only call this when fetchDepositAddress?

Nope, we should implement fetchCurrencies and check for credentials in fetchCurrencies, for example, here:

- https://github.com/ccxt/ccxt/blob/master/js/okx.js#L1050

**Pull request merged**